specs/Print.md

```diff
@@ -0,0 +1,464 @@
+  /// If page range is not valid or if a page is greater than document total page count,
+  /// `ICoreWebView2PrintWithSettingsCompletedHandler` handler will return `E_ABORT`.
+  ///
+  /// |       Example       | Valid |               Notes                   |
+  /// "2"                   |  Yes  | Assuming document total page count >= 2.
+  /// "1-4, 9, 3-6, 10, 11" |  Yes  | Assuming document total page count >= 11.
+  /// "1-4, -6"             |  Yes  | Assuming document total page count >= 6.
+  /// "2-"                  |  Yes  | Assuming document total page count >= 2, means from 2 to the * end.
+  /// "4-2, 11, -6"         |  No   |
+  ///  "-"                  |  Yes  | Assuming document total page count >= 1.
+  ///  "1-4dsf, 11"         |  No   | Regardless of document total page count.
+  ///
+  /// The caller must free the returned string with `CoTaskMemFree`. See
+  /// [API Conventions](/microsoft-edge/webview2/concepts/win32-api-conventions#strings)
+  [propget] HRESULT PageRanges([out, retval] LPWSTR* value);
+
+  /// Set the `PageRanges` property.
+  [propput] HRESULT PageRanges([in] LPCWSTR value);
+
+  /// Number of pages per sheet. The Default value is 1 and maximum is 16.
+  [propget] HRESULT PagesPerSheet([out, retval] INT32* value);
+
+  /// Set the `PagesPerSheet` property. Returns `E_INVALIDARG` if an invalid value is
+  /// provided, and the current value is not changed.
+  [propput] HRESULT PagesPerSheet([in] INT32 value);
+
+  /// Number of copies to print. The Default value is 1 and maximum
+  /// copies count is 999.
+  /// This value is ignored in PrintToPdfStream method.
+  [propget] HRESULT Copies([out, retval] INT32* value);
+
+  /// Set the `Copies` property. Returns `E_INVALIDARG` if an invalid value is provided
+  /// for the specified printer, and the current value is not changed.
+  [propput] HRESULT Copies([in] INT32 value);
+
+  /// True if the printed document should be collated. The default value is `FALSE`.
+  /// This value is ignored in PrintToPdfStream method.
+  [propget] HRESULT ShouldCollate([out, retval] BOOL* value);
+
+  /// Set the `ShouldCollate` property.
+  [propput] HRESULT ShouldCollate([in] BOOL value);
+
+  /// True if the print should be in color, otherwise prints in black and white.
+  /// The default value is `TRUE`.
+  [propget] HRESULT IsColor([out, retval] BOOL* value);
+
+  /// Set the `IsColor` property.
+  [propput] HRESULT IsColor([in] BOOL value);
+
+  /// Printer's duplex settings. See `COREWEBVIEW2_PRINT_SIDE` for descriptions
+  /// of print sides. The default value is `COREWEBVIEW2_PRINT_SIDE_SINGLE`.
+  /// This property is ignored by the PrintToPdf and PrintToPdfStream methods.
+  [propget] HRESULT PrintSide([out, retval] COREWEBVIEW2_PRINT_SIDE* value);
+
+  /// Set the `PrintSide` property.
+  [propput] HRESULT PrintSide([in] COREWEBVIEW2_PRINT_SIDE value);
+
+  /// The horizontal printer resolution for the page, in dots per inch.
+  /// The default value is 600.
```

Comment:
I didn't find minimum and maximum values for quality to include.

Below are the values I see upon querying printer capabilities (native API) and print preview dialog uses these values to display qualities to the end user.

**File printers: **

Microsoft PDF - Only 600 
OneNote for Desktop - Only 600
XPS - Only 600
OneNotes for Windows - Only 300

**Physical printer: **
Hp printer on my machine - Default is 600 but dpi's it supports are 300/600/1200

**Fax:**
Fax - 200

However, upon manually hard coding dpi's values in our code, printing doesn't throw any error for values I have tested with file printers ( -600, 0, 450, 150).

Removed about invalid values statement here but I am not really sure how to address the qualities since these values varies and we don't know when we create the printSettings.

As printing API doesn't throw any error, should we simply pass the value user has provided without validation? What should be the default value? Here I added 600 but what if end user selects 
Should we provide Quality property in our API?  This not asked by our partner, but I added this to match with the edge print dialog since we are saying this API helps for custom print dialog and not sure any of our GitHub customers would be interested in this. 

Do you have any other suggestions?

Reply:
At least we can document that the allowed values for Quality vary from printer to printer and that end devs should consult OS printer APIs to determine allowed values?

What happens when you provide the default 600 to OneNote which only allows 300? It would be good to document what happens when an invalid value is provided.

Reply:
Chromium code prints with the printer default quality if we provide an invalid value. In WebView2, we check provided qualities for the provided printer and run completed handler with an error if it is not in the printer qualities list.

Added the doc accordingly.